wolfcrypt/src/rsa.c

```diff
@@ -1287,24 +1303,27 @@ static int RsaPrivateDecryptEx(byte* in, word32 inLen, byte* out,
         /* fall through */
     }
     case RSA_STATE_DECRYPT_RES:
-        key->state = RSA_STATE_NONE;
-    #if defined(WOLFSSL_ASYNC_CRYPT) && defined(HAVE_CAVIUM)
+        key->state = RSA_STATE_DECRYPT_RES;
+    #if defined(WOLFSSL_ASYNC_CRYPT) && defined(WC_ASYNC_ENABLE_RSA) && \
+            defined(HAVE_CAVIUM)
         if (key->asyncDev.marker == WOLFSSL_ASYNC_MARKER_RSA) {
-            ret = key->tmpLen;
+            /* return event ret */
+            ret = key->asyncDev.event.ret;
         }
     #endif
         break;
     default:
         ret = BAD_STATE_E;
     }
 
+done:
```

Comment:
Can remove and change ERROR_OUT set ret and break.
Confusing to have multiple ways out of switch.